### PR TITLE
Add database_account azure metricset

### PR DESCRIPTION
**Pull request by narph**
Should solve https://github.com/elastic/beats/issues/15758

Lightweight metricset based on `monitor`

Config:
```
  - module: azure
    metricsets:
    - database_account
    enabled: true
    period: 300s
    client_id: '${AZURE_CLIENT_ID:""}'
    client_secret: '${AZURE_CLIENT_SECRET:""}'
    tenant_id: '${AZURE_TENANT_ID:""}'
    subscription_id: '${AZURE_SUBSCRIPTION_ID:""}'
```
extra config options as in the other metricsets:

```
resources:
 - resource_group: []
 - resource_id: []
```
The metrics in https://docs.microsoft.com/en-us/azure/azure-monitor/platform/metrics-supported#microsoftdocumentdbdatabaseaccounts should be retrieved.
The `DatabaseName` dimension is applied for the metrics which support it.

cc: @sorantis

### CODE REVIEW

File: metricbeat/docs/modules/azure.asciidoc [hunk 1]
```diff
@@ -149,6 +149,17 @@ metricbeat.modules:
   client_secret: '${AZURE_CLIENT_SECRET:""}'
   tenant_id: '${AZURE_TENANT_ID:""}'
   subscription_id: '${AZURE_SUBSCRIPTION_ID:""}'
+
+  - module: azure
```
Comment (kaiyan-sheng) [inline metricbeat/docs/modules/azure.asciidoc]: indentation is wrong here, this section needs to be moved to left.

File: x-pack/metricbeat/module/azure/database_account/_meta/data.json [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    "metricset" : {
+        "name" : "database_account",
+        "period" : 300000
+    },
+    "azure" : {
+        "dimensions" : {
+            "databasename" : "testdb"
+        },
+        "database_account" : {
+            "timegrain" : "PT5M",
```
Comment (exekias) [inline x-pack/metricbeat/module/azure/database_account/_meta/data.json]: is this field mapped by `fields.yml`?, we may need to add an entry

File: x-pack/metricbeat/module/azure/database_account/manifest.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
+default: true
+input:
+  module: azure
+  metricset: monitor
```
Comment (exekias) [inline x-pack/metricbeat/module/azure/database_account/manifest.yml]: :heart: to see this!

File: x-pack/metricbeat/module/azure/database_account/_meta/data.json [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+            "provisionedthroughput" : {
```
Comment (exekias) [inline x-pack/metricbeat/module/azure/database_account/_meta/data.json]: Are we grouping metrics with the same dimensions & resource in a single document? That would save some space and make queries faster

File: x-pack/metricbeat/module/azure/database_account/_meta/data.json (hunk 2 above)
Comment (exekias) [reply x-pack/metricbeat/module/azure/database_account/_meta/data.json]: perhaps we should move it to `azure.timegrain`? as it's common to all metricsets

Comment (narph) [reply x-pack/metricbeat/module/azure/database_account/_meta/data.json]: indeed, I have moved `timegrain` inside the `azure` object as this is a common property for all metricsets

File: x-pack/metricbeat/module/azure/database_account/_meta/data.json (hunk 4 above)
Comment (narph) [reply x-pack/metricbeat/module/azure/database_account/_meta/data.json]: yes, all metrics with the same timegrain, dimension and resource are grouped in the same event

File: x-pack/metricbeat/module/azure/_meta/fields.yml [hunk 5]
```diff
@@ -8,6 +8,10 @@
       type: group
       description: >
       fields:
+        - name: timegrain
+          type: keyword
+          description: >
+            The azure metric timegrain
```
Comment (exekias) [inline x-pack/metricbeat/module/azure/_meta/fields.yml]:
```suggestion
            The Azure metric timegrain
```

File: x-pack/metricbeat/metricbeat.reference.yml [hunk 6]
```diff
@@ -265,6 +265,17 @@ metricbeat.modules:
...
 
+  - module: azure
+    metricsets:
+    - database_account
+    enabled: true
+    period: 300s
+    client_id: '${AZURE_CLIENT_ID:""}'
+    client_secret: '${AZURE_CLIENT_SECRET:""}'
+    tenant_id: '${AZURE_TENANT_ID:""}'
+    subscription_id: '${AZURE_SUBSCRIPTION_ID:""}'
```
Comment (kaiyan-sheng) [inline x-pack/metricbeat/metricbeat.reference.yml]:
```suggestion
- module: azure
  metricsets:
  - database_account
  enabled: true
  period: 300s
  client_id: '${AZURE_CLIENT_ID:""}'
  client_secret: '${AZURE_CLIENT_SECRET:""}'
  tenant_id: '${AZURE_TENANT_ID:""}'
  subscription_id: '${AZURE_SUBSCRIPTION_ID:""}'
```

Comment (narph) [reply x-pack/metricbeat/metricbeat.reference.yml]: @kaiyan-sheng this was fixed in the earlier commits

**Pull request merged**